onnxruntime/core/providers/qnn/builder/opbuilder/pool_op_builder.cc

```diff
@@ -98,6 +106,9 @@ Status PoolOpBuilder::SetCommonPoolParams(const NodeAttrHelper& node_helper,
 
   std::vector<int32_t> pads = {0, 0, 0, 0};
   auto auto_pad = node_helper.Get("auto_pad", std::string("NOTSET"));
+  ORT_RETURN_IF(auto_pad != "NOTSET" && auto_pad != "SAME_LOWER" && auto_pad != "SAME_UPPER",
+                "QNN Pool operators do not support 'auto_pad' value: ", auto_pad.c_str());
+
```

Comment:
Fixes test errors that snuck in while Linux QNN Pipeline did not run unit tests. See https://github.com/microsoft/onnxruntime/pull/15497